responsibleai/responsibleai/rai_insights/rai_insights.py

```diff
@@ -586,14 +586,22 @@ def _validate_rai_insights_input_parameters(
                 "identified as categorical features: "
                 f"{non_categorical_or_time_string_columns}")
 
+        # Check if any of the data is missing in test and train data
         list_of_feature_having_missing_values = []
         for feature in test.columns.tolist():
             if np.any(test[feature].isnull()):
                 list_of_feature_having_missing_values.append(feature)
         if len(list_of_feature_having_missing_values) > 0:
-            warnings.warn(
+            raise UserConfigValidationException(
                 f"Features {list_of_feature_having_missing_values} "
                 "have missing values in test data")
+        for feature in train.columns.tolist():
+            if np.any(train[feature].isnull()):
+                list_of_feature_having_missing_values.append(feature)
```

Comment:
This looks like repeated code. Could you create a common function for this?

Reply:
Also, looks like you are using the same list for both train and test. Though functionally the logic is correct it can easily break. Creating a common function will resolve this problem.

Reply:
@vinuthakaranth for the bug: https://msdata.visualstudio.com/Vienna/_workitems/edit/2349068 you might want to check here and see if this handles the case of NaN's. The error comes from this https://scikit-learn.org/stable/modules/generated/sklearn.utils.assert_all_finite.html checking for NaNs and the null check here may not be covering that.

Reply:
@Hannah, can you please check if NaN case is also handled with the same fix as it is the similar error? If not, I'll reopen the bug and follow up.

Reply:
@vinuthakaranth it does! 
@gaugup see my comment below about moving this check for test data and let me know your thoughts